.github/workflows/build_and_test.yaml

```diff
@@ -12,68 +12,53 @@ on:
       - "release-v*"
     paths-ignore:
       - "**/*.png"
-env:
-  GO_VERSION: 1.18.2
-  ENVOY_GATEWAY_DEV_IMAGE: envoyproxy/gateway-dev
-  ENVOY_GATEWAY_DEV_TAG: latest
 jobs:
+  lint:
+    runs-on: ubuntu-latest
```

Comment:
we'd like to test and build on the same runner

Reply:
is there a specific reason to run them separately rather than on the same runner and changing the existing workflow ?

Reply:
1. It keeps it clear what dependencies there are between the different things happening in the job.  As the workflow grows, it can become difficult to untangle a single monolithic job that might have grown hidden dependencies between steps within it.
2. Faster. Time spent waiting on CI is down to ~1m from ~2m.
3. Higher quality feedback. "You munged the whitespace on line X" shouldn't stop you from getting feedback on "does the thing actually work"